Make testtool run out-of-the-box in ooenv-ed env

From: Thorsten Behrens <thb@openoffice.org>


---

 automation/source/testtool/objtest.cxx  |   63 +++++++++++++++++++++++++++++++
 automation/source/testtool/testtool.ini |    1 
 solenv/bin/linkoo                       |    1 
 3 files changed, 64 insertions(+), 1 deletions(-)


diff --git automation/source/testtool/objtest.cxx automation/source/testtool/objtest.cxx
index dceb0fa..f3f6e33 100644
--- automation/source/testtool/objtest.cxx
+++ automation/source/testtool/objtest.cxx
@@ -531,6 +531,69 @@ void TestToolObj::LoadIniFile()				// Laden der IniEinstellungen, die durch den
     GetHostConfig();
     GetTTPortConfig();
     GetUnoPortConfig();
+
+    aConf.SetGroup("Crashreporter");
+
+    String aUP;
+    GETSET( aUP, "UseProxy", "false" );
+    String aPS;
+    GETSET( aPS, "ProxyServer", "" );
+    String aPP;
+    GETSET( aPP, "ProxyPort", "" );
+    String aAC;
+    GETSET( aAC, "AllowContact", "false" );
+    String aRA;
+    GETSET( aRA, "ReturnAddress", "" );
+
+    OUString sPath;
+    if( osl_getExecutableFile( (rtl_uString**)&sPath ) == osl_Process_E_None)
+    {
+        sPath = sPath.copy(7); // strip file://
+
+        int i = sPath.lastIndexOf('/');
+        if (i >= 0)
+            i = sPath.lastIndexOf('/', i-1 );
+
+        if (i >= 0)
+        {
+            sPath = sPath.copy(0, i);
+            ByteString bsPath( sPath.getStr(), sPath.getLength(),
+                               RTL_TEXTENCODING_UTF8 );
+            
+            aConf.SetGroup( "OOoProgramDir" );
+            String aOPD;
+            // testtool is installed in Basis3.x/program/ dir nowadays
+            bsPath += "/../program";
+            GETSET( aOPD, "Current", bsPath);
+
+            ByteString aSrcRoot;
+            aSrcRoot += getenv("SRC_ROOT");
+            aConf.SetGroup( "_profile_Default" );
+            if (aSrcRoot.Len())
+            {
+                String aPBD;
+                aSrcRoot += "/testautomation";
+                GETSET( aPBD, "BaseDir", aSrcRoot );
+
+                String aPHD;
+                aSrcRoot += "/global/hid";
+                GETSET( aPHD, "HIDDir", aSrcRoot );
+            }
+            else
+            {
+                String aPBD;
+                bsPath += "/qatesttool";
+                GETSET( aPBD, "BaseDir", bsPath );
+                
+                String aPHD;
+                bsPath += "/global/hid";
+                GETSET( aPHD, "HIDDir", bsPath );
+            }
+            
+            String aLD;
+            GETSET( aLD, "LogBaseDir", ByteString( "/tmp" ) );
+        }
+    }
 }
 
 #define MAKE_TT_KEYWORD( cName, aType, aResultType, nID )                       \
diff --git automation/source/testtool/testtool.ini automation/source/testtool/testtool.ini
index bf1b54c..58fa92e 100644
--- automation/source/testtool/testtool.ini
+++ automation/source/testtool/testtool.ini
@@ -9,7 +9,6 @@ CurrentProfile=_profile_Default
 
 [OOoProgramDir]
 Type=Path
-Current=.
 
 [Crashreporter]
 UseProxy=false
diff --git solenv/bin/linkoo solenv/bin/linkoo
index 659d473..a4615d5 100755
--- solenv/bin/linkoo
+++ solenv/bin/linkoo
@@ -372,6 +372,7 @@ if (!-f "$OOO_INSTALL/" . $brand_program_dir . "/ooenv") {
     print "Creating '$OOO_INSTALL/", $brand_program_dir, "/ooenv'\n";
     open ($ooenv, ">$OOO_INSTALL/" . $brand_program_dir . "/ooenv") || die "Can't open $OOO_INSTALL/" . $brand_program_dir . "/ooenv: $!";
     print $ooenv $env_script;
+	print $ooenv "# testtool\nexport SRC_ROOT=$OOO_BUILD\n";
     close ($ooenv);
 }
 
